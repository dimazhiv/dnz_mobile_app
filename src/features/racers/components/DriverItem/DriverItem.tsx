import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { useOpenLink } from '../../../../hooks/hooks';

export type DriverItemProps = {
  driverId: string;
  url: string;
  fullName: string;
  dateOfBirth: string;
  nationality: string;
  //TODO DZ correct types
  onPress: any;
};

function DriverItem({ driverId, url, fullName, dateOfBirth, nationality, onPress }: DriverItemProps) {
  const handlePress = () => onPress(driverId);
  const handleLinkPress = useOpenLink(url);

  return (
    <View style={styles.container}>
      <View style={styles.nameCell}>
        <TouchableOpacity onPress={handlePress}>
          <Text style={styles.name}>{fullName}</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.dateCell}>
        <Text>{dateOfBirth}</Text>
      </View>
      <View style={styles.countryCell}>
        <Text numberOfLines={1} ellipsizeMode={'tail'}>
          {nationality}
        </Text>
      </View>
      <View style={styles.racesCell}>
        <TouchableOpacity onPress={handleLinkPress}>
          <Text style={styles.linkStyle} numberOfLines={2}>
            info
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    borderColor: 'lightgrey',
    alignItems: 'center',
    flexDirection: 'row'
  },
  nameCell: {
    padding: 10,
    width: 100
  },
  dateCell: {
    padding: 10,
    width: 110
  },
  countryCell: {
    padding: 10,
    width: 100
  },
  racesCell: {
    padding: 10,
    minWidth: 60
  },
  name: {
    color: 'blue',
    fontWeight: 'bold'
  },
  linkStyle: {
    color: 'blue',
    textDecorationLine: 'underline'
  }
});

export default DriverItem;
