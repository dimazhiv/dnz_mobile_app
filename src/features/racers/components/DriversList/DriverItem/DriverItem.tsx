import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';

export type DriverItemProps = {
  driverId: string;
  fullName: string;
  dateOfBirth: string;
  nationality: string;
  onPress: (id: string) => void;
  onRacesPress: (id: string) => void;
};

function DriverItem({ driverId, fullName, dateOfBirth, nationality, onPress, onRacesPress }: DriverItemProps) {
  const handlePress = () => onPress(driverId);
  const handleRacesPress = () => onRacesPress(driverId);

  return (
    <View style={styles.container}>
      <View style={styles.nameCell}>
        <TouchableOpacity onPress={handlePress}>
          <Text style={styles.name}>{fullName}</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.dateCell}>
        <Text>{dateOfBirth}</Text>
      </View>
      <View style={styles.countryCell}>
        <Text numberOfLines={1} ellipsizeMode={'tail'}>
          {nationality}
        </Text>
      </View>
      <View style={styles.racesCell}>
        <TouchableOpacity onPress={handleRacesPress}>
          <Text style={styles.linkStyle} numberOfLines={2}>
            races
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    borderColor: 'lightgrey',
    alignItems: 'center',
    flexDirection: 'row'
  },
  nameCell: {
    padding: 10,
    width: 100
  },
  dateCell: {
    padding: 10,
    width: 110
  },
  countryCell: {
    padding: 10,
    width: 100
  },
  racesCell: {
    padding: 10,
    minWidth: 60
  },
  name: {
    color: 'blue',
    fontWeight: 'bold'
  },
  linkStyle: {
    color: 'blue',
    textDecorationLine: 'underline'
  }
});

export default DriverItem;
