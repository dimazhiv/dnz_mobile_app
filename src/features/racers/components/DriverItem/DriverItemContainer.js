import DriverItem from './DriverItem';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import {getDriver} from "../../selectors";

function mapStateToProps(state, ownProps) {
    const {
        driverId,
        url,
        givenName,
        familyName,
        dateOfBirth,
        nationality
    } = getDriver(state, ownProps.driverId) || {};

    return {
        driverId,
        url,
        fullName: `${givenName} ${familyName}`,
        dateOfBirth,
        nationality

    }
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators(
        {},
        dispatch
    );
}

export default connect(mapStateToProps, mapDispatchToProps)(DriverItem);
