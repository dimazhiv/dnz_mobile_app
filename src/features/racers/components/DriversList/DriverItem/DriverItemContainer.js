import DriverItem from './DriverItem';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { getDriver } from '../../../selectors';
import { onDriverNamePress, onRacesPress } from '../../../sagaActions';

function mapStateToProps(state, ownProps) {
  const { driverId, url, givenName, familyName, dateOfBirth, nationality } = getDriver(state, ownProps.driverId) || {};

  return {
    driverId,
    url,
    fullName: `${givenName} ${familyName}`,
    dateOfBirth,
    nationality
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      onPress: onDriverNamePress,
      onRacesPress: onRacesPress
    },
    dispatch
  );
}

export default connect(mapStateToProps, mapDispatchToProps)(DriverItem);
