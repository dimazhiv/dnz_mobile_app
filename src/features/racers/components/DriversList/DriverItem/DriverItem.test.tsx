import React from 'react';
import { shallow } from 'enzyme';
import DriverItem from './DriverItem';

describe('Snapshot testing DriverItem component', () => {
  test('correct component view', () => {
    const tree = shallow(
      <DriverItem
        dateOfBirth={'01.09.1994'}
        fullName={'Carlos Sainz'}
        nationality={'Spanish'}
        driverId={'alonso'}
        onPress={() => {}}
        onRacesPress={() => {}}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
