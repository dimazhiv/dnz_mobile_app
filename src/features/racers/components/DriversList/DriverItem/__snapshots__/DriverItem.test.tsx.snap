// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing DriverItem component correct component view 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderColor": "lightgrey",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      {
        "padding": 10,
        "width": 100,
      }
    }
  >
    <TouchableOpacity
      onPress={[Function]}
    >
      <Text
        style={
          {
            "color": "blue",
            "fontWeight": "bold",
          }
        }
      >
        Carlos Sainz
      </Text>
    </TouchableOpacity>
  </View>
  <View
    style={
      {
        "padding": 10,
        "width": 110,
      }
    }
  >
    <Text>
      01.09.1994
    </Text>
  </View>
  <View
    style={
      {
        "padding": 10,
        "width": 100,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
    >
      Spanish
    </Text>
  </View>
  <View
    style={
      {
        "minWidth": 60,
        "padding": 10,
      }
    }
  >
    <TouchableOpacity
      onPress={[Function]}
    >
      <Text
        numberOfLines={2}
        style={
          {
            "color": "blue",
            "textDecorationLine": "underline",
          }
        }
      >
        races
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;
